File: 1262.js
const fs = require('fs');

const input = fs.readFileSync(0, 'utf8').trim().split(/\s+/);

let idx = 0;

while (idx < input.length) {
    const trace = input[idx++];
    const procs = parseInt(input[idx++], 10);

    let ciclos = 0;
    let i = 0;
    const n = trace.length;

    while (i < n) {
        if (trace[i] === 'W') {
            ciclos++;
            i++;
        } else { 
            let qtdR = 0;
            while (i < n && trace[i] === 'R') {
                qtdR++;
                i++;
            }
            ciclos += Math.ceil(qtdR / procs);
        }
    }

    console.log(ciclos);
}

File: 1262.java
import java.lang.Math;
import java.util.Scanner;

public class Main {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		String trace;
		int procs;
				
		while(sc.hasNext()) {
			trace = sc.next();
			procs = sc.nextInt();
			
			int conta, i, tam;
			conta = 0;
			i = 0;
			tam = trace.length();
			
			while(i < tam) {
				if (trace.charAt(i) == 'W') {
	                conta++;
	                i++;
	            } else { // 'R'
	                int contaR = 0;
	                while (i < tam && trace.charAt(i) == 'R') {
	                    contaR++;
	                    i++;
	                }
	                conta += (int)Math.ceil((double)contaR / procs);
	            }
			}
			
			System.out.print(conta + "\n");
		}
		
			
		sc.close();
	}
}